accepted/2024/package-update-command.md

```diff
@@ -0,0 +1,286 @@
+  --mode <direct|promote>  The implementation will have a description here to explain the differences between direct and promote.
+  --project                Path to a project or solution file, or a directory.
+  --vulnerable             Update packages with a known vulnerability.
+  -?, -h, --help           Show help and usage information
+```
+
+### Scenarios
+
+#### Works with other features
+
+This probably can go without saying, but it's expected that all `dotnet package update` commands work with all other Nuget features.
+
+For example, projects that use Central Package Management should have versions updated in the Directory.Packages.props file.
+Another example is when a nuget.config has enabled Package Source Mapping, the updated packages must take into account the versions available on the enabled sources, so that subsequent restores will be successful.
+
+Similarly, since `dotnet package update` is part of the .NET SDK, it's only expected to work with projects that work with `dotnet build`.
+Therefore, some Visual Studio projects will not work, where they can't be built with the dotnet CLI.
+
+#### Updating all packages to the latest version
+
+Running `dotnet package update` without specifying any packages, or options such as `--vulnerable`, will automatically update all packages to the highest version.
+
+```text
+C:\> dotnet package update
+
+Fixing outdated packages in ContosoUniversity.sln
+ ContosoLibrary:
+  Updated UmbracoForms 8.4.1 to 8.7.1
+  Updated Newtonsoft.Json 11.0.1 to 13.0.3
+
+ ContosoApp:
+  Updated UmbracoForms 8.4.0 to 8.7.1
+
+Updated 3 packages in 36 scanned packages.
+```
+
+#### Fixing vulnerabilities
+
+When users run `dotnet build` or `dotnet restore` commands in CLI, if they see any warnings related to vulnerabilities in their project’s NuGet dependencies, they can run `dotnet package update --vulnerable` CLI command to try to remediate all the vulnerabilities.
+The `--mode` option can be used to choose between direct package references only, or to also update transitive packages.
+
+```CLI
+C:\ContosoApp\> dotnet package update --vulnerable
+
+Fixing vulnernable packages in ContosoUniversity.sln
+ ContosoLibrary:
+  Upgrading UmbracoForms 8.4.1 to 8.7.1
+
+ ContosoApp:
+  Updated UmbracoForms 8.4.0 to 8.7.1
+    
+Fixed 2 packages in 36 scanned packages.
+```
+
+Using the `--vulnerable` option should update the package to the lowest version that is higher than the currently reference version, which does not have a known vulnerability.
+For example, if `Contoso.Security` version 1.0.0 is currently referenced by the project, and the package source has versions 1.0.1 and 2.0.0 available, 1.0.1 should be selected.
+This is in contrast to `dotnet package update Contoso.Security` which will update to the highest version of the package.
+
+If package identifiers are provided in addition to the `--vulnerable` option, only those packages should be updated.
```

Comment:
Feels like we're missing `--packages` option from the description.

Reply:
packages are an argument, not an option. For example `dotnet package update pkg1 pkg2 pkg3 --vulnerable`.